crengine/src/private/lvfreetypeface.cpp

```diff
@@ -828,19 +843,18 @@ bool LVFreeTypeFace::hbCalcCharWidth(LVCharPosInfo *posInfo, const LVCharTriplet
             // which will be the one that will be rendered
             FT_UInt ch_glyph_index = FT_Get_Char_Index( _face, triplet.Char );
             if ( glyph_info[cluster].codepoint == ch_glyph_index ) {
-                posInfo->offset = FONT_METRIC_TO_PX(glyph_pos[cluster].x_offset);
-                posInfo->width = FONT_METRIC_TO_PX(glyph_pos[cluster].x_advance);
+                posInfo->offset_26_6 = glyph_pos[cluster].x_offset;
+                posInfo->advance_26_6 = glyph_pos[cluster].x_advance;
                 return true;
             }
         }
-
     }
-    // Otherwise, use plain Freetype getGlyphInfo() which will check
+    // Otherwise, use plain Freetype getGlyph() which will check
     // again with this font, or the fallback one
-    glyph_info_t glyph;
-    if ( getGlyphInfo(triplet.Char, &glyph, def_char, fallbackPassMask) ) {
-        posInfo->offset = 0;
-        posInfo->width = glyph.width;
+    LVFontGlyphCacheItem *glyph = getGlyph(triplet.Char, def_char, fallbackPassMask);
```

Comment:
getGlyphInfo() should be faster than getGlyph(), as it just get and cache metrics, while getGlyph loads and caches the glyph bitmap. Might make full rendering slower with HB light.
Any reason to really switch to getGlyph (other than not wanting to hack getGlyphInfo ?)

Reply:
Firstly, this is a fallback branch, which is not always work, secondly, getGlyphInfo() rounds all sizes to an integer but getGlyph() returns advance in fixed-point format 26.6. In this way, we achieve the best possible match between HarfBuzz light and HarfBuzz.